web/controllers/account_controller.ex

```diff
@@ -1,14 +1,48 @@
 defmodule Entice.Web.AccountController do
   use Entice.Web.Web, :controller
   alias Entice.Web.Account
+  alias Entice.Web.Invitation
+  alias Entice.Web.Queries
+  require Logger
+
+  plug :ensure_login when action in [:request_invite]
 
   def create(conn, _params) do
     email = conn.params["email"]
     password = conn.params["password"]
+    invite_key = conn.params["inviteKey"]
+
+    case Queries.get_invite(email) do
+      
+      {:error, :no_matching_invite} ->
+        conn |> json error(%{message: "No Invitation found for this Email"})
+
+      {:ok, invite} ->
+        if invite.key != invite_key do
```

Comment:
Put this in a guard clause within the match instead, or match on the same invite key:
Option A)
```Elixir
  {:ok, invite} when invite.key != invite_key -> #error here
  {:ok, invite} -> #else clause here
 ```
Option B)
```Elixir
  {:ok, %Invitation{key: ^invite_key} = invite} -> #else clause here
  {:ok, _invitation} -> #error clause here
```
Obviously, the second is more cryptic and communicates less the intent than the first one.
(The second matches the structure of the Invitation itself and defines that the key value must be the same as ```invite_key```, because of the caret ```^``` - check elixir docs [here](http://elixir-lang.org/getting-started/case-cond-and-if.html))